protostar/commands/test/environments/fuzz_test_execution_environment.py

```diff
@@ -0,0 +1,190 @@
+import asyncio
+import contextvars
+import functools
+import inspect
+import re
+from dataclasses import dataclass
+from typing import Optional, List, Callable, Awaitable, Any, Dict
+
+from hypothesis import settings, seed, given, Verbosity
+from hypothesis.database import InMemoryExampleDatabase
+from hypothesis.reporting import with_reporter
+from hypothesis.strategies import data, DataObject
+
+from protostar.commands.test.environments.test_execution_environment import (
+    TestExecutionEnvironment,
+    TestCaseCheatcodeFactory,
+)
+from protostar.commands.test.fuzzing.fuzz_input_exception_metadata import (
+    FuzzInputExceptionMetadata,
+)
+from protostar.commands.test.fuzzing.strategy_selector import StrategySelector
+from protostar.commands.test.starkware.execution_resources_summary import (
+    ExecutionResourcesSummary,
+)
+from protostar.commands.test.starkware.test_execution_state import TestExecutionState
+from protostar.commands.test.test_context import TestContextHintLocal
+from protostar.commands.test.test_environment_exceptions import ReportedException
+from protostar.commands.test.testing_seed import current_testing_seed
+from protostar.utils.data_transformer_facade import DataTransformerFacade
+
+HYPOTHESIS_VERBOSITY = Verbosity.normal
+"""
+Change this value to ``Verbosity.verbose`` while debugging Hypothesis adapter code.
+"""
+
+
+def is_fuzz_test(function_name: str, state: TestExecutionState) -> bool:
+    abi = state.contract.abi
+    params = DataTransformerFacade.get_function_parameters(abi, function_name)
+    return bool(params)
+
+
+class FuzzTestExecutionEnvironment(TestExecutionEnvironment):
+    def __init__(self, state: TestExecutionState):
+        super().__init__(state)
+
+    async def invoke(self, function_name: str) -> Optional[ExecutionResourcesSummary]:
+        abi = self.state.contract.abi
+        parameters = DataTransformerFacade.get_function_parameters(abi, function_name)
+        assert (
+            parameters
+        ), f"{self.__class__.__name__} expects at least one function parameter."
```

Comment:
Maybe fuzzing should enforce parameters naming convention. 
We may try to pass `__setup__`'s `context` keys as function parameters in the future.

Reply:
1. This assertion only protects from bogus `invoke_test` implementation. Do you think it's work adding explanation comment here?
2. I don't see a need for any naming conventions regarding parameters. Moreover I believe this would be unergonomic for users, as parameters will be locals users will want to use in their tested expressions.